### PR TITLE
system-variables: add system variable tidb_gc_max_wait_time

### PR DESCRIPTION
**Pull request by TonsnakeLin**
Signed-off-by: TonsnakeLin <lpbgytong@163.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add system variable tidb_gc_max_wait_time

### Which TiDB version(s) do your changes apply to? (Required)
V6.1

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:[#32726](https://github.com/pingcap/tidb/pull/32726)
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1032,6 +1032,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 >     - 大量的历史数据可能会在一定程度上影响系统性能，尤其是范围的查询（如 `select count(*) from t`）。
 > - 如果一个事务的运行时长超过了 `tidb_gc_life_time` 配置的值，在 GC 时，为了使这个事务可以继续正常运行，系统会保留从这个事务开始时间 `start_ts` 以来的数据。例如，如果 `tidb_gc_life_time` 的值配置为 10 分钟，且在一次 GC 时，集群正在运行的事务中最早开始的那个事务已经运行了 15 分钟，那么本次 GC 将保留最近 15 分钟的数据。
 
+### `tidb_gc_max_wait_time` <span class="version-mark">从 v6.1 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`86400`
+- 范围：`[600, 31536000]`
+- 这个变量用于指定活跃事务阻碍 GC Safe Point 推进的最大时间。变量值为整型值，单位是秒。每次进行 GC 时，如果活跃事务运行时间未超过该值， GC Safe Point 一直会被阻塞，直到活跃事务运行时间超过该值 Safe Point 才会正常往前推进。
```
Comment (sticnarf) [inline system-variables.md:1041]: 可能加上相关背景更容易理解，像这样？

```suggestion
- 这个变量用于指定活跃事务阻碍 GC Safe Point 推进的最大时间。变量值为整型值，单位是秒。每次进行 GC 时，默认 GC safe point 不会超过正在执行中的事务的开始时间。如果活跃事务运行时间未超过该值， GC Safe Point 一直会被阻塞不更新，直到活跃事务运行时间超过该值 Safe Point 才会正常往前推进。
```

**Review by sticnarf — APPROVED** [review]

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

File: system-variables.md [hunk 2]
```diff
@@ -1032,6 +1032,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 >     - 大量的历史数据可能会在一定程度上影响系统性能，尤其是范围的查询（如 `select count(*) from t`）。
 > - 如果一个事务的运行时长超过了 `tidb_gc_life_time` 配置的值，在 GC 时，为了使这个事务可以继续正常运行，系统会保留从这个事务开始时间 `start_ts` 以来的数据。例如，如果 `tidb_gc_life_time` 的值配置为 10 分钟，且在一次 GC 时，集群正在运行的事务中最早开始的那个事务已经运行了 15 分钟，那么本次 GC 将保留最近 15 分钟的数据。
 
+### `tidb_gc_max_wait_time` <span class="version-mark">从 v6.1 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:1035]:
```suggestion
### `tidb_gc_max_wait_time` <span class="version-mark">从 v6.1.0 版本开始引入</span>
```

Comment (Oreoxmt) [inline system-variables.md:1035]:
```suggestion
### `tidb_gc_max_wait_time` <span class="version-mark">从 v6.1.0 版本开始引入</span>
```

**Review by Oreoxmt — APPROVED** [review]

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**